tap_s3_csv/sync.py

```diff
@@ -113,14 +113,18 @@ def handle_file(config, s3_path, table_spec, stream, extension, file_handler=Non
             LOGGER.info('using S3 Get Range method for csv import')
             # csv.DictReader will parse the first non-empty row as header if fieldnames == None, else as the first record.
             # For parallel threads, non-first threads will not be able to grab headers from the first part of the data,
-            # so we need to pass in fieldnames. First thread needs to handle first row in order to avoid having first row parsed
-            # as record when it's actually header. Set handle_first_row param for PreprocessStream to True for first thread.
-            file_handle = preprocess.PreprocessStream(file_handle, table_spec, start_byte == 0)
+            # so we need to pass in fieldnames. First thread needs to handle first row if table_spec.has_header == True in order to avoid 
+            # having first row parsed as record when it's actually header. Set handle_first_row param for PreprocessStream to True
+            # for this case so that the file/stream pointer is moved to skip first row.
+            file_handle = preprocess.PreprocessStream(file_handle, table_spec, start_byte == 0 and table_spec.get('has_header', True))
```

Comment:
we don't need to handle first row if has_header == false. csv.DictReader will simply take fieldnames passed in and use it as header and parse first row as record. If has_header == true, handle_first_row function is simply ran to move fp to point second row so that header row is not included.